addons/godot-xr-tools/objects/grab_points/grab.gd

```diff
@@ -169,3 +180,51 @@ func _clear_hand_pose() -> void:
 
 	# Remove hand snapping
 	hand.remove_target_override(hand_point)
+
+
+# Add collision exceptions for the grabbed object and any of its children
+func _add_collision_exceptions(from : Node):
+	if not is_instance_valid(collision_hand):
+		return
+
+	if not is_instance_valid(from):
+		return
+
+	# If this is a physics body, add an exception
+	if from is PhysicsBody3D:
+		print_debug("Add collision exception for ", from.name)
+		# Make sure we don't collide with what we're holding
+		_collision_exceptions.push_back(from)
+		collision_hand.add_collision_exception_with(from)
+		from.add_collision_exception_with(collision_hand)
+
+	# Check all children
+	for child in from.get_children():
+		_add_collision_exceptions(child)
+
+
+# Remove the exceptions in our passed array. We call this with a small delay
+# to give an object a chance to drop away from the hand before it starts
+# colliding.
+# It is possible that another object is picked up in the meanwhile
+# and we thus fill _collision_exceptions with new content.
+# Hence using a copy of this list at the time of dropping the object.
+#
+# Note, this is static because our grab object gets destroyed before this code gets run.
+static func _remove_collision_exceptions( \
+	on_collision_hand : XRToolsCollisionHand, \
+	exceptions : Array[PhysicsBody3D]):
+	if not is_instance_valid(on_collision_hand):
+		return
+
+	# This can be improved by checking if we're still colliding and only
```

Comment:
This is also for a follow up PR. It would require manually checking collisions on the physics server. Would be a cool improvement but for later.